pilot/pkg/model/context.go

```diff
@@ -948,3 +948,7 @@ func (node *Proxy) GetInterceptionMode() TrafficInterceptionMode {
 
 	return InterceptionRedirect
 }
+
+// SidecarDNSListenerPort specifes the port at which the sidecar hosts a DNS resolver listener.
+// TODO: customize me
```

Comment:
I may have mentioned this: my preference is to use a 'basePort' - and have offsets for all other ports.
We can use the admin port as base (15000). 

If the base port is 0, all ports should be dynamic (testing) - and we need to figure a way to consistently get the ports.